x-pack/plugins/security_solution/cypress/test_plans/detection_response/prebuilt_rules/installation_and_upgrade.md

```diff
@@ -0,0 +1,580 @@
+- `should update outdated prebuilt rules when previous historical versions unavailable`
+
+Notes:
+
+- Legacy API:
+  - install: `PUT /api/detection_engine/rules/prepackaged`
+  - upgrade: `PUT /api/detection_engine/rules/prepackaged`
+  - status: `GET /api/detection_engine/rules/prepackaged/_status`
+- New API:
+  - install: `POST /internal/detection_engine/prebuilt_rules/installation/_perform`
+  - upgrade: `POST /internal/detection_engine/prebuilt_rules/upgrade/_perform`
+  - status: `GET /internal/detection_engine/prebuilt_rules/status`
+
+#### **Scenario: API does not install or upgrade prebuilt rules if they are up to date**
+
+**Automation**: 4 integration tests with mock rules.
+
+```Gherkin
+Given the package <package_type> is installed
+And the package contains N rules
+When user installs all rules via install <api>
+And user gets prebuilt rules status via status <api>
+Then the endpoint should return 200 with <status_response>
+When user calls install <api>
+Then the endpoint should return 200 with <install_response>
+When user calls upgrade <api>
+Then the endpoint should return 200 with <upgrade_response>
+
+Examples:
+  | package_type             | api    | status_response                  | install_response         | upgrade_response         |
+  | with historical versions | legacy | not_installed: 0, not_updated: 0 | installed: 0, updated: 0 | installed: 0, updated: 0 |
+  | w/o historical versions  | legacy | not_installed: 0, not_updated: 0 | installed: 0, updated: 0 | installed: 0, updated: 0 |
+  | with historical versions | new    | to_install: 0, to_upgrade: 0     | total: 0, succeeded: 0   | total: 0, succeeded: 0   |
+  | w/o historical versions  | new    | to_install: 0, to_upgrade: 0     | total: 0, succeeded: 0   | total: 0, succeeded: 0   |
+```
+
+Notes:
+
+- Legacy API:
+  - install: `PUT /api/detection_engine/rules/prepackaged`
+  - upgrade: `PUT /api/detection_engine/rules/prepackaged`
+  - status: `GET /api/detection_engine/rules/prepackaged/_status`
+- New API:
+  - install: `POST /internal/detection_engine/prebuilt_rules/installation/_perform`
+  - upgrade: `POST /internal/detection_engine/prebuilt_rules/upgrade/_perform`
+  - status: `GET /internal/detection_engine/prebuilt_rules/status`
+
+### Rule installation and upgrade notifications on the Rule Management page
+
+#### **Scenario: User is notified when no prebuilt rules are installed**
+
+**Automation**: 1 e2e test with mock rules + 1 integration test.
+
+```Gherkin
+Given no prebuilt rules are installed in Kibana
+And there are X prebuilt rules available to install
+When user opens the Rule Management page
+Then user should see a CTA to install prebuilt rules
+And user should see a number of rules available to install (X)
+And user should NOT see a CTA to upgrade prebuilt rules
+And user should NOT see a number of rules available to upgrade
+And user should NOT see the Rule Updates table
```

Comment:
THis is not the current behaviour: Rules Updates table is always visible and accesible, but if there are now updates available a message is displayed explaining that everythings is up-to date.

Reply:
Should we change that behaviour? I can create a ticket but not sure such a fix would be able to get done for 8.9 it's a feature change.

Reply:
Behaviour changed in https://github.com/elastic/kibana/pull/161196